app/models/apply.rb

```diff
@@ -1,6 +1,6 @@
 class Apply < ApplicationRecord
   has_many :pet_apply
-  has_many :pets, through: :pet_apply
+  has_many :pets, through: :pet_apply, dependent: :destroy
```

Comment:
Good catch here!

Reply:
This was a toughy - I should have trusted Ruby, she told me that it was connected to the join table and that's why it couldn't destroy!